### PR TITLE
feat(complete): complete sigil-prefixed positionals

### PR DESCRIPTION
**Pull request by jdx**
## Summary
- match completions against the value after the declared sigil
- restore the sigil on emitted completion candidates
- keep explicit double dash behavior intact
- add dedicated sigil documentation and completion examples

## Stack
- Depends on #1322

## Test plan
- cargo test -p usage-cli complete_word_sigil
- cargo test -p usage-conformance --test sigil
- mise run lint

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

---

> [!NOTE]
> **Medium Risk**
> Completion routing changes are broad (argv, CLI, overlays, file fallback, restart/`--` edge cases) and user-visible at the shell, though scoped to completion with new tests.
> 
> **Overview**
> **Tab completion now understands arguments declared with a `sigil` prefix** (for example `+node@22`). When the current word starts with that prefix, matchers use the text *after* the sigil and put the prefix back on every candidate. The same strip/restore behavior applies to static choices, declared completion types, and runtime/async overlays.
> 
> **Sigil fields are treated as overlays on the positional cursor**, not the next ordinary slot. Restart handling (`:::`), default-subcommand guessing, file fallback, and overlay targeting all use a new **`first_ordinary_arg`** helper so leading sigil-only args are skipped. **`sigil_arg_at_cursor`** (and the parallel logic in `complete_word`) picks the longest active sigil; sigil completion is disabled after `--`, when a flag is awaiting a value, or once a restart token has appeared earlier on the line. Explicit `--` still prevents classifying `+…` as a sigil.
> 
> The **usage CLI `cw` path** mirrors the argv engine: `run=` completers see the stripped current word via an adjusted Tera `words` context, and **`flags_possible`** also turns off after an `automatic` double-dash trailing segment. **Docs and tests** add a sigils reference page, an example spec, and coverage for choices, runtime completion, restart, and separator behavior.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 3d27ebdb6f9dc2b669159c2c5ce11e6c9b5982ff. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: argv/src/complete.rs [hunk 1]
```diff
@@ -1426,6 +1426,7 @@ fn overlay_at_cursor<'o>(
         return None;
     }
     let meta = metadata_chain_on_route(spec, position).and_then(|chain| chain.last().copied());
+    let sigil_target = sigil_arg_at_cursor(spec, position, &split.prefix);
```
[the remark this answers is not in the document]
Comment (jdx) [reply argv/src/complete.rs]: Fixed in 66218eae33. Cursor-field resolution now prefers the matching sigil argument for both closed-choice and file-fallback decisions; the regression test covers an ordinary file positional beside a closed sigil field.\n\n*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

[the remark this answers is not in the document]
Comment (jdx) [reply argv/src/complete.rs]: Fixed in 66218eae33. Runtime completion callbacks now receive the stripped sigil suffix, filter on that suffix, and have the sigil restored on emitted candidates. The same behavior is applied to named completion overlays and covered by the sigil test.\n\n*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

File: argv/src/complete.rs [hunk 2]
```diff
@@ -1191,7 +1192,9 @@ fn complete_inner<'a>(
     let at_cursor = if after_restart {
         meta.and_then(|m| m.args.first()).map(|m| m.arg)
     } else {
-        position.next_arg
+        sigil_arg_at_cursor(spec, &position, token)
+            .map(|(_, field, _, _)| field.arg)
+            .or(position.next_arg)
```
[the remark this answers is not in the document]
Comment (jdx) [reply argv/src/complete.rs:1202]: Fixed in e469a24388: any active sigil cursor now suppresses raw shell file fallback, including explicitly file-typed sigil fields and dynamic completion paths. Added a regression for a path-typed open sigil argument.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

File: cli/src/cli/complete_word.rs [hunk 3]
```diff
@@ -219,6 +234,13 @@ impl CompleteWord {
             let (found, closed) = self.complete_arg(&cx, &parsed.cmd, arg, &ctoken)?;
             has_explicit_choices = closed || arg.choices.is_some();
             found
+        } else if let Some((arg, sigil, prefix)) = sigil_arg {
+            let (mut found, closed) = self.complete_arg(&cx, &parsed.cmd, arg, prefix)?;
+            for (candidate, _) in &mut found {
+                candidate.insert_str(0, sigil);
+            }
+            has_explicit_choices = closed || arg.choices.is_some() || found.is_empty();
+            found
```
[the remark this answers is not in the document]
Comment (jdx) [reply cli/src/cli/complete_word.rs:287]: Fixed in 5609fcec31: CLI completion now derives the current flag state after automatic trailing arguments and disables sigil classification permanently after a restart, matching the parser.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

File: argv/src/complete.rs [hunk 4]
```diff
@@ -1496,6 +1535,29 @@ fn overlay_at_cursor<'o>(
...
+            let sigil = field
+                .arg
+                .sigil
+                .and_then(|value| core::str::from_utf8(value).ok())?;
+            token
+                .strip_prefix(sigil)
+                .map(|prefix| (owner, field, sigil, prefix))
+        })
+        .max_by_key(|(_, _, sigil, _)| sigil.len())
+}
```
[the remark this answers is not in the document]
Comment (jdx) [reply argv/src/complete.rs:1559]: Fixed in 5609fcec31: argv sigil cursor selection now declines when a detached flag value is pending or the cursor is immediately after a restart. Those contexts retain their own candidates and file fallback; regressions cover both.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

File: cli/src/cli/complete_word.rs [hunk 5]
```diff
@@ -219,6 +261,13 @@ impl CompleteWord {
...
```
[the remark this answers is not in the document]
Comment (jdx) [reply cli/src/cli/complete_word.rs:287]: Fixed in 683bbed6f0: sigil matching now retains the owning command and passes it to complete_arg, so ancestor-declared completion blocks are evaluated in their own command context.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

File: argv/src/complete.rs [hunk 6]
```diff
@@ -1496,6 +1533,31 @@ fn overlay_at_cursor<'o>(
...
```
[the remark this answers is not in the document]
Comment (jdx) [reply argv/src/complete.rs:1559]: Fixed in `5399c01c7a`: argv completion now scans all prior words for the active command's restart token, so sigils stay disabled for the remainder of the invocation; the regression covers a later sigil-shaped token.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*

File: cli/src/cli/complete_word.rs [hunk 7]
```diff
@@ -219,6 +261,13 @@ impl CompleteWord {
             let (found, closed) = self.complete_arg(&cx, &parsed.cmd, arg, &ctoken)?;
             has_explicit_choices = closed || arg.choices.is_some();
             found
+        } else if let Some((owner, arg, sigil, prefix)) = sigil_arg {
+            let (mut found, closed) = self.complete_arg(&cx, owner, arg, prefix)?;
+            for (candidate, _) in &mut found {
+                candidate.insert_str(0, sigil);
+            }
+            has_explicit_choices = closed || arg.choices.is_some() || found.is_empty();
+            found
```
[the remark this answers is not in the document]
Comment (jdx) [reply cli/src/cli/complete_word.rs:287]: Fixed in `895f5f3cf9`: sigil completion now supplies the stripped current token through `words[CURRENT]` while rendering dynamic `run=` completers, then restores the sigil only on returned candidates. Added an end-to-end regression using a `run=` template that echoes the current word; focused completion tests and Rust 1.94 Clippy pass.\n\n_This reply was AI-generated by Codex._

File: argv/src/complete.rs [hunk 8]
```diff
@@ -1188,10 +1193,13 @@ fn complete_inner<'a>(
     // argument, whatever the words before the token filled, and everything below follows from
     // that: whether paths belong, whether the set is declared, whether a separator is owed.
     let after_restart = restarted(meta, split);
+    let sigil_cursor = sigil_arg_at_cursor(spec, &position, split);
```
[the remark this answers is not in the document]
Comment (jdx) [reply argv/src/complete.rs:1196]: Fixed in 3d27ebdb6f. Restart completion now skips leading sigil-only arguments everywhere the cursor target is consumed: candidates, file fallback, overlays, and traces. The same cursor rule now applies to default-subcommand completion, with compiled clause targets preserved downstream. Added compiled and reference completion regressions.

_This reply was AI-generated by Codex._
